### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by Samuellekpor**
In this feature branch, we achieved the following:

- Install CanCanCan in your project.
- Add a `role` column to the users table. 
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). 
    - Implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`).
    - Implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
     @user = User.find(params[:user_id])
-    @posts = @user.posts.includes(:comments)
+    if current_user.nil?
+      redirect_to new_user_session_path
+      return
+    end
+    @posts = if current_user.role?
+               @user.posts.includes(:comments)
+             else
+               @user.posts.includes(:comments, :author)
```
Comment (jerryowusu) [inline app/controllers/posts_controller.rb:11]: Kindly consider adding `load_and_authorize_resource` in the Post Controller in order to automatically set the authorizations for the controller. By calling `load_and_authorize_resource` on a controller, most of the CRUD method calls on models are added automatically. :wink:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @Samuellekpor & @AbayomiOlaoye 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost-paul-walker


## Highlights
No linter errors :heavy_check_mark: 
Added cancancan to the gemfile :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Samuellekpor) [reply app/controllers/posts_controller.rb:11]: Hello @jerryowusu!!!

Thanks for the review! I would like to ask if this is a project requirement? I didn't see it anywhere and i managed to do it manually. Changing everything and adding `load_and_authorize_resource` will delay my progress. Is the app working now on your side or not please?

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @Samuellekpor, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**